apps/microtvm/zephyr/demo_runtime/src/main.c

```diff
@@ -153,113 +190,132 @@ tvm_crt_error_t TVMPlatformTimerStop(double* elapsed_time_seconds) {
   // if we approach the limits of the HW clock datatype (uint32_t), use the
   // coarse-grained timer result instead
   if (approx_num_cycles > (0.5 * (~((uint32_t)0)))) {
-    *elapsed_time_seconds = timer_res_ms / 1e3;
+    *res_us = timer_res_ms * 1000.0;
   } else {
-    *elapsed_time_seconds = hw_clock_elapsed_seconds;
+    *res_us = hw_clock_res_us;
   }
 
   g_utvm_timer_running = 0;
   return kTvmErrorNoError;
 }
 
-tvm_crt_error_t TVMPlatformGenerateRandom(uint8_t* buffer, size_t num_bytes) {
-  uint32_t random;  // one unit of random data.
+// Memory pool for use by TVMPlatformMemoryAllocate.
+K_MEM_POOL_DEFINE(tvm_memory_pool, 64, 1024, 216, 4);
 
-  // Fill parts of `buffer` which are as large as `random`.
-  size_t num_full_blocks = num_bytes / sizeof(random);
-  for (int i = 0; i < num_full_blocks; ++i) {
-    random = sys_rand32_get();
-    memcpy(&buffer[i * sizeof(random)], &random, sizeof(random));
-  }
-
-  // Fill any leftover tail which is smaller than `random`.
-  size_t num_tail_bytes = num_bytes % sizeof(random);
-  if (num_tail_bytes > 0) {
-    random = sys_rand32_get();
-    memcpy(&buffer[num_bytes - num_tail_bytes], &random, num_tail_bytes);
-  }
+// Used by TVM to allocate memory.
+tvm_crt_error_t TVMPlatformMemoryAllocate(size_t num_bytes, DLContext ctx, void** out_ptr) {
+  *out_ptr = k_mem_pool_malloc(&tvm_memory_pool, num_bytes);
+  return (*out_ptr == NULL) ? kTvmErrorPlatformNoMemory : kTvmErrorNoError;
+}
 
+// Used by TVM to deallocate memory.
+tvm_crt_error_t TVMPlatformMemoryFree(void* ptr, DLContext ctx) {
+  k_free(ptr);
   return kTvmErrorNoError;
 }
 
-#define RING_BUF_SIZE 512
+// Ring buffer used to store data read from the UART on rx interrupt.
+#define RING_BUF_SIZE 20 * 1024
```

Comment:
optional: this should really be size_bytes, but looking [more](https://docs.zephyrproject.org/2.4.0/reference/kernel/other/ring_buffers.html?highlight=ring%20buffers#byte-mode) it seems as if we may not be declaring the ring buf correctly. can you either fix or add a TODO(areusch) to that?